client/state/sites/actions.js

```diff
@@ -115,13 +115,22 @@ export function setFrontPage( siteId, pageId, successCallback ) {
 			page_on_front_id: pageId,
 		};
 
-		return wpcom.undocumented().setSiteHomepageSettings( siteId, requestData ).then( () => {
+		return wpcom.undocumented().setSiteHomepageSettings( siteId, requestData ).then( ( response ) => {
+			const updatedOptions = {
+				page_on_front: parseInt( response.page_on_front_id, 10 ),
+				show_on_front: response.is_page_on_front ? 'page' : 'posts',
+			};
+
+			if ( response.page_for_posts_id ) {
```

Comment:
Even if `page_for_posts_id` is zero it should be set. In fact, this is what happens when I try and verify the fix... setting the posts page to the homepage results in this being zero.